sidecar/proxy/envoy/manager.go

```diff
@@ -403,15 +408,84 @@ func BuildWeightKey(service string, tags []string) string {
 	return fmt.Sprintf("%v.%v", service, BuildServiceKey("_", tags))
 }
 
+func buildCluster(clusterName string, backend *api.Backend, tlsConfig *SSLContext) Cluster {
+	cluster := Cluster{
+		Name:             clusterName,
+		ServiceName:      clusterName,
+		Type:             "sds",
+		LbType:           "round_robin",
+		ConnectTimeoutMs: 1000,
+		CircuitBreakers:  &CircuitBreakers{},
+		OutlierDetection: &OutlierDetection{
+			MaxEjectionPercent: 100,
+		},
+		SSLContext: tlsConfig,
+	}
+
+	if backend != nil && backend.LbType != "" {
+		// Set default value of LbType to be "round_robin"
+		cluster.LbType = backend.LbType
+	}
+
+	if backend.Resilience != nil {
+		// Cluster level settings
+		if backend.Resilience.MaxRequestsPerConnection > 0 {
+			cluster.MaxRequestsPerConnection = backend.Resilience.MaxRequestsPerConnection
+		}
+
+		// Envoy Circuit breaker config options
+		if backend.Resilience.MaxConnections > 0 {
+			cluster.CircuitBreakers.MaxConnections = backend.Resilience.MaxConnections
+		}
+		if backend.Resilience.MaxRequests > 0 {
+			cluster.CircuitBreakers.MaxRequests = backend.Resilience.MaxRequests
+		}
+		if backend.Resilience.MaxPendingRequest > 0 {
+			cluster.CircuitBreakers.MaxPendingRequest = backend.Resilience.MaxPendingRequest
+		}
+
+		// Envoy outlier detection settings that complete circuit breaker
+		if backend.Resilience.SleepWindow > 0 {
+			cluster.OutlierDetection.BaseEjectionTimeMS = int(backend.Resilience.SleepWindow * 1000)
+		}
+		if backend.Resilience.ConsecutiveErrors > 0 {
+			cluster.OutlierDetection.ConsecutiveError = backend.Resilience.ConsecutiveErrors
+		}
+		if backend.Resilience.DetectionInterval > 0 {
+			cluster.OutlierDetection.IntervalMS = int(backend.Resilience.DetectionInterval * 1000)
+		}
+	}
+
+	return cluster
+}
+
 // BuildRoutes builds routes based on rules.  Assumes at least one route for each service
 func BuildRoutes(ruleList []api.Rule, instances []*api.ServiceInstance) []Route {
 	SanitizeRules(ruleList)
 	rules := AddDefaultRouteRules(ruleList, instances)
 
+	staticClusters := make(map[string]*api.Backend)
```

Comment:
We never use the map value `*api.Backend`, so `staticClusters` would probably be more appropriate as a set: `map[string]struct{}`